### ISSUE TITLE
Sample request: Activity DI

### ISSUE THREAD

**Issue by cretz** [temporalio/samples-dotnet#9]
### Describe the solution you'd like

People need to access their DI-based objects inside an activity and it may not be obvious how to do this manually.

May also be related to https://github.com/temporalio/sdk-dotnet/issues/46

**Comment by cretz** [comment]
Can do the approach taken by https://github.com/devbased/temporal-activity-gen but with activity interceptor. Could also have constructor based injection supported as well with the class accepting activity classes and just reflectively finding methods.

**Comment by mrtristan** [comment]
Following

just landed in exactly this position. @cretz - Looks like the mentioned predecessor PR was merged so is this now possible?

**Comment by cretz** [comment]
Yes, I just need to write the sample. I think it's just gonna use the new `ActivityDefinition.Create` with a custom invoker that will use the service provider to create the activity instance on each invocation.

**Comment by mrtristan** [comment]
sounds like it makes sense - my only question at the moment is where you're getting the service provider from. can that be DI'd? What's the scope?

**Comment by dystopiandev** [comment]
Constructor injection for activities would make perfect sense.
We would just have to add activities to the service collection.

I say this without a clear understanding of why that could be difficult to achieve.
@cretz could you ELI5 the limitations?
